### PR TITLE
Add Showback model

### PR DESCRIPTION
**Pull request by yaborens**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json [hunk 1]
```diff
@@ -1688,7 +1688,85 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+	"/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.CostManagement/Showback/{ruleName}": {
```
Comment (jianghaolu) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: Fix indentation

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json [hunk 2]
```diff
@@ -1688,7 +1688,85 @@
         }
       }
     },
-    "/providers/Microsoft.CostManagement/operations": {
+	"/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.CostManagement/Showback": {
+      "get": {
+        "tags": [
+          "Showback"
+        ],
+        "operationId": "ShowbackRules_List",
```
Comment (jianghaolu) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: Is this a paging operation?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json (hunk 1 above)
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: ok

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json (hunk 2 above)
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: no

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json [hunk 3]
```diff
@@ -1,2872 +1,3102 @@
...
-          "CloudConnectors"
-        ],
-        "operationId": "CloudConnector_Delete",
-        "description": "Delete a cloud connector definition",
-        "externalDocs": {
-          "url": "https://docs.microsoft.com/en-us/rest/api/costmanagement/"
-        },
-        "x-ms-examples": {
-          "CloudConnector_Delete": {
-
```
Comment (KrisBash) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: please change to `showbackRules` as the resource Type name for internal consistency and proper naming patterns

Comment (KrisBash) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: `id` here is redundant, as the resource name is contained in the top-level schema. Please remove

Comment (KrisBash) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: `readOnly` here seems spurious

Comment (KrisBash) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: There are potential PII/compliance concerns with a user name property. Please remove from this PR, or contact me offline with approvals for this.

Comment (KrisBash) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: Defining this as an open object prevents client tooling from validating/assisting proper input. Have a look at [this swagger](https://github.com/Azure/azure-rest-api-specs/blob/672c412811930cefa3c02143f342ce493f7b154e/specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/externalSecuritySolutions.json) to see use of the "discriminator" concept. In essence, you define a "discriminator" property based on the enum `ruleType` and define three definitions for this object. On the new object definitions, you would set a `x-ms-discriminator-valuex-ms-discriminator-value` property to match one of the three enum values.

**Review by KrisBash — CHANGES_REQUESTED** [review]
Thanks for the review meeting. My notes are captured in comments on this PR

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json (hunk 3 above)
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: I changed it to ShowbackRules

Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: I removed modificationUsername

Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: I removed ID

Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: I removed Id

Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: done

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json [hunk 4]
```diff
@@ -1652,10 +1652,10 @@
 				}
 			}
 		},
-		"/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.CostManagement/Showback": {
+		"/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.CostManagement/ShowbackRules": {
```
Comment (KrisBash) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: minor comment: resource type names should be lowerCamelCase, i.e. .`showbackRules`

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json [hunk 5]
```diff
@@ -2966,11 +2940,131 @@
...
+					"$ref": "#/definitions/CustomChargeProperties"
+				}
+			}
+		},
+		"CustomChargeProperties": {
+			"type": "object",
+			"title": "Custom charge showback rule properties",
+			"properties": {
+				"recurring": {
+					"type": "boolean"
```
Comment (KrisBash) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: A string enum with values of "true" and "false" is a more future proof type than boolean. Changing type in the future is a breaking change

**Review by KrisBash — APPROVED** [review]
Signing off with comments .Please review

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json (hunk 4 above)
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: ok, i changed it

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json (hunk 5 above)
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: ok, i changed it to enum

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json (hunk 4 above)
Comment (yaborens) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: done

**Comment by yaborens** [comment]
@jianghaolu 
Kris from ARM team approved the PR.
Can we please continue and get your blessing?

**Review by jianghaolu — APPROVED** [review]

**Pull request merged**